crates/core/database/src/models/users/model.rs

```diff
@@ -297,23 +302,28 @@ impl User {
             }
         }
 
-        // Ensure none of the following substrings show up in the username
-        const BLOCKED_SUBSTRINGS: &[&str] = &[
-            "```",
-            "discord.gg",
-            "rvlt.gg",
-            "guilded.gg",
-            "https://",
-            "http://",
-        ];
-
-        for substr in BLOCKED_SUBSTRINGS {
-            if username_lowercase.contains(substr) {
-                return Err(create_error!(InvalidUsername));
-            }
+        Ok(())
+    }
+
+    /// sanitise username
+    fn sanitise_username(username: &str) -> (bool, String) {
+        let original_username = username;
+        // sanitise homoglyphs
+        let options = decancer::Options::default()
+            .retain_capitalization();
+        let mut username = decancer::cure(&username, options).unwrap().to_string();
+
+        if BLOCKED_USERNAME_PATTERNS.is_match(&username) {
+            username = BLOCKED_USERNAME_PATTERNS.replace_all(&mut username, "").into_owned();
         }
```

Comment:
Running the regex twice seems redundant. replace_all has no performance hit if a match isn't found.